src/middlewared/middlewared/plugins/system.py

```diff
@@ -1359,11 +1401,35 @@ def set_crash_reporting(self):
         CrashReporting.enabled_in_settings = self.middleware.call_sync('system.general.config')['crash_reporting']
 
 
+# Update Birthday Date
+async def _update_birthday(middleware):
+    # Sync clock
+    middleware.logger.debug('Synchornization the clock for system birthday')
+    birthday = None
+    timeout = 60
+
+    while birthday is None:
+        birthday = await middleware.call('system.sync_clock')
+        # Wait until be able to sync the clock
+        if birthday is None:
+            await asyncio.sleep(timeout)
```

Comment:
Dont think we should retry every 60 seconds, this seems overly aggressive. Perhaps only once a day, and at most 5 retries or so?

We could probably also put it in a hook to retry after network changes are made (e.g. interface.post_sync)

Reply:
Added both as suggested.